clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitJobExecutor.java

```diff
@@ -41,6 +42,7 @@
 public class GitJobExecutor {
 
   private static final String SSH_KEY_PWD_ENV_VAR = "SSH_KEY_PWD";
+  private static final Pattern SHA_PATTERN = Pattern.compile("[0-9a-f]{40}");
```

Comment:
Should we also support shorter SHA (like the default 7 characters)?

Reply:
I actually couldn't find official docs from jgit on how it supports cloning SHA. The method used before only talks about ref name, branch name or tag name: https://download.eclipse.org/jgit/site/5.10.0.202012080955-r/apidocs/org/eclipse/jgit/api/CloneCommand.html#setBranch-java.lang.String-

Besides `git fetch` only supporting the full SHA, I'll check with the user that reported the issue.

Reply:
+1 on supporting shorter SHAs. As a matter of fact, we have an internal Run Job stage that depends on Clouddriver/Git repo support and short SHAs so preserving this behavior would be great.

Reply:
That's an interesting problem, we cannot use short SHAs in git fetch commands:
```
git fetch origin 9904d5e
fatal: couldn't find remote ref 9904d5e
```
It's possible that the only way to make short SHAs to work is to actually download the entire repo (with full history) and then checkout that SHA.

Reply:
I added support for short SHAs as a fallback mechanism in case regular cloning of branch/tag doesn't work, this is to account for cases where a branch or tag matches the same pattern of a short SHA. The full repo needs to be cloned for this to work though.